### PR TITLE
Prevent zero-priced CFD offers

### PR DESCRIPTION
**Pull request by klochowicz**
Simplify the UI code and disable the button for posting a new offer if there is
no price set.

### CODE REVIEW

File: frontend/src/MakerApp.tsx [hunk 1]
```diff
@@ -187,18 +185,6 @@ export default function App() {
     );
 }
 
-function priceToDisplay(hasEnteredPrice: boolean, orderPrice: string, priceInfo: PriceInfo | null) {
-    if (!priceInfo) {
-        return format("0");
-    }
-
-    if (!hasEnteredPrice) {
-        return format((priceInfo.ask * SPREAD).toString());
```
Comment (thomaseizinger) [inline frontend/src/MakerApp.tsx:196]: If you delete this, we are no longer calculating a default price using the spread.

File: frontend/src/MakerApp.tsx [hunk 2]
```diff
@@ -187,18 +185,6 @@ export default function App() {
...
-    }
-
-    return format(orderPrice);
```
Comment (thomaseizinger) [inline frontend/src/MakerApp.tsx:199]: We are no longer calling `format` on the price now, is that intentional?

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Two questions.

File: frontend/src/MakerApp.tsx (hunk 1 above)
Comment (klochowicz) [reply frontend/src/MakerApp.tsx:196]: there was no default price using the spread. It was displayed, but not used anywhere... that was the bug in question.

Comment (klochowicz) [reply frontend/src/MakerApp.tsx:196]: is there a way of triggering the "refresh" button when the `priceInfo` becomes available? That would solve the actual problem in a much nicer way IMHO.

File: frontend/src/MakerApp.tsx (hunk 2 above)
Comment (klochowicz) [reply frontend/src/MakerApp.tsx:199]: fixed (as per the fixup commit) - added format when the button refreshes.

File: frontend/src/MakerApp.tsx [hunk 3]
```diff
@@ -107,19 +112,18 @@ export default function App() {
...
-                                    if (priceInfo) {
-                                        setOrderPrice((priceInfo.ask * SPREAD).toString());
-                                    }
+                                    setAutoRefresh(false);
                                 }}
+                                value={format(orderPrice)}
                             />
+                            <HStack>
+                                <Switch
+                                    id="auto-refresh"
```
Comment (thomaseizinger) [inline frontend/src/MakerApp.tsx:121]: This can go away if you are not using it for attaching a label to the component.

**Review by thomaseizinger — APPROVED** [review]

**Comment by thomaseizinger** [comment]
At some point it would be nice to invest some time into UI component tests as well :)

**Review by da-kami — APPROVED** [review]
noice!

File: frontend/src/MakerApp.tsx (hunk 3 above)
Comment (klochowicz) [reply frontend/src/MakerApp.tsx:121]: I find it still useful for conveying the purpose of the switch (it's self-explanatory now, without going into the details of the callback)

**Pull request merged**